0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -232,61 +250,77 @@ type
+Support for a condition type MUST depend on one or more feature suites.  However, all new condition types MUST depend on at least one of the new feature suites. This ensures that all previously created implementations correctly recognize that they do not support the new type.
 
 Feature suites are chosen such that they represent reasonable clusters of functionality. For instance, it is reasonable to require that an implementation which supports SHA-256 in one context MUST support it in all contexts, since it already needed to implement the algorithm.
 
 An implementation which supports a certain set of feature suites MUST accept all condition types which depend only on that set or any subset of feature suites.
 
+| Suite Name | Feature Bit | Feature Bit (BASE16) | Summary |
+|------------|-----|------|---------------------------------|
+| SHA-256    | 2^0 | 0x01 | The SHA-256 hashing algorithm. |
+| PREIMAGE   | 2^1 | 0x02 | The functionality of comparing a hash to a preimage. |
+| PREFIX     | 2^2 | 0x04 | The functionality of prefixing the fulfillment with a prefix before generating the condition. |
+| THRESHOLD  | 2^3 | 0x08 | The functionality of composing a condition out of several weighted subconditions. |
+| RSA-PSS    | 2^4 | 0x10 | The RSA-PSS signature algorithm. |
+| ED25519    | 2^5 | 0x20 | The ED25519 signature algorithm. |
+
 ## SHA-256 {#sha-256-feature-suite}
-SHA-256 is a hashing algorithm and is assigned the feature bit 2^0 = 0x01.
 
-## PREIMAGE {#preimage-feature-suite}
-PREIMAGE refers to hashlock conditions and is assigned the feature bit 2^1 = 0x02.
+SHA-256 is a cryptographic hash function published by the US National Institute of Standards and Technology that produces 256 bits of output. This feature suite is assigned the feature bit 2^0 = 0x01.
 
-A preimage condition is the hash of its own fulfillment. In order to fulfill a preimage condition, a valid preimage must be provided.
+## PREIMAGE {#preimage-feature-suite}
+PREIMAGE refers to conditions that use a preimage as a one-time signature. This feature suite is assigned the feature bit 2^1 = 0x02.
 
-Preimage conditions can be used as a so-called hashlock. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can feasibly produce the preimage if it contains a large amount of random entropy.
+The fingerprint of a preimage condition is the hash of some arbitrary value. The payload of a preimage fulfillment is the preimage that hashes to the condition's fingerprint. Conditions that use this preimage MUST also rely on a cryptographically secure hashing algorithm. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can produce the preimage, as long as it contains a large amount of random entropy.
 
```

Comment:
It's not obvious from this paragraph that the "arbitrary value" and the payload of the fullfilment are the same thing.